src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -127,11 +127,11 @@ class IpexCommunicationService extends AgentService {
     }
     const pickedCred = creds[0];
     let holderSignifyName;
-    const holder = await this.identifierStorage.getIdentifierMetadata(
-      msg.exn.a.i
-    );
-    if (holder && holder.signifyName) {
-      holderSignifyName = holder.signifyName;
+    const signifyName = (
+      await this.identifierStorage.getIdentifierMetadata(msg.exn.a.i)
+    ).signifyName;
+    if (signifyName) {
+      holderSignifyName = signifyName;
```

Comment:
We can get rid of the `holderSignifyName` and the if statement. `signifyName` is always set so we can just continue straight away with:

```
    const signifyName = (
      await this.identifierStorage.getIdentifierMetadata(msg.exn.a.i)
    ).signifyName;
    const [offer, sigs, gend] = await this.signifyClient.ipex().grant
```